<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>435</width>
    <height>511</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>spass</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tabPassword">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <attribute name="title">
        <string>Password</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QCheckBox" name="checkBox">
          <property name="text">
           <string>Uppercase</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="strip" stdset="0">
           <string notr="true">ABCDEFGHIJKLMNOPQRSTUVWXYZ</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBox_3">
          <property name="text">
           <string>Numbers</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="strip" stdset="0">
           <string>0123456789</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBox_4">
          <property name="text">
           <string>Special Charactres</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="strip" stdset="0">
           <string notr="true">!@#$%^&amp;*_-</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBox_2">
          <property name="text">
           <string>Lowercase</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="strip" stdset="0">
           <string notr="true">abcdefghijklmnopqrstuvwxyz</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabPassphrase">
       <attribute name="title">
        <string>Passphrase</string>
       </attribute>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>45</x>
          <y>26</y>
          <width>281</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Using Default wordlist</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Length</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QSlider" name="horizontalSlider_length">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>20</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="lengthSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>65536</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Stength</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QProgressBar" name="strengthBar">
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="format">
         <string>%v bits</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="generateButton">
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="outputEdit">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="copyButton">
      <property name="text">
       <string>Copy</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;2013 (C) Guy Rutenberg&lt;br/&gt;&lt;a href=&quot;http://guyrutenberg.com/spass&quot;&gt;&lt;span style=&quot;color:#0000ff;&quot;&gt;http://guyrutenberg.com/spass&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;Source Code: &lt;a href=&quot;http://github.com/guyru/spass&quot;&gt;&lt;span style=&quot;color:#0000ff;&quot;&gt;http://github.com/guyru/spass&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QDialogButtonBox" name="buttonBox">
      <property name="standardButtons">
       <set>QDialogButtonBox::Close</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>435</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>checkBox</tabstop>
  <tabstop>outputEdit</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>generateButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_length</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lengthSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>185</y>
    </hint>
    <hint type="destinationlabel">
     <x>388</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lengthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_length</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lengthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrength()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>432</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>currentChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>194</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>261</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateStrip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateStrength()</slot>
  <slot>updateStrip()</slot>
 </slots>
</ui>
